### PR TITLE
Feature/7731 show recurring cost row

### PR DESCRIPTION
**Pull request by bjss-chris**
Story: [AB#7731](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7731)
Task: 7884

image

### CODE REVIEW

File: app/pages/preview/contextCreator.js [hunk 1]
```diff
@@ -58,7 +58,73 @@ const generateCallOffAndSupplierDetailsTable = ({
...
 
-export const getContext = ({ orderId, orderData }) => ({
+const generateRecurringCostDetailsTable = ({
+  recurringCostTable, recurringCostItems = [], serviceRecipients = {},
+}) => {
+  const items = recurringCostItems.map((item) => {
+    const classes = 'nhsuk-u-font-size-14';
+    const columns = [];
+
+    if (!Object.prototype.hasOwnProperty.call(serviceRecipients, item.serviceRecipientsOdsCode)) {
```
Comment (alibombaye) [inline app/pages/preview/contextCreator.js:68]: think you can shorthand this too

`if (!serviceRecipients[item.serviceRecipientsOdsCode])`

File: app/pages/preview/contextCreator.js [hunk 2]
```diff
@@ -58,7 +58,73 @@ const generateCallOffAndSupplierDetailsTable = ({
...
+      throw new Error(`service recipient ${item.serviceRecipientsOdsCode} not found`);
```
Comment (alibombaye) [inline app/pages/preview/contextCreator.js:69]: The pattern we have used in the app for throwing errors is something like this

```
logger.error(`blah blah blah`);
throw new Error();
```

File: app/pages/preview/helpers/getServiceRecipients.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const getServiceRecipients = (serviceRecipients = []) => {
```
Comment (alibombaye) [inline app/pages/preview/helpers/getServiceRecipients.js:1]: just on the name of this function. Any reason you went for `getServiceRecipients`? we normally use the `get` prefix when it's going to call the api to get some data. Do you think we could rename this function to something else?

Comment (bjss-chris) [reply app/pages/preview/helpers/getServiceRecipients.js:1]: I wasn't aware that you reserved the `get` prefix for API calls and I just thought that `getServiceRecipients` was more concise than `createServiceRecipientsDict`. However, I'll rename to the latter.

**Pull request merged**